test/VaultAsyncMode.spec.ts

```diff
@@ -0,0 +1,335 @@
+import { depositParams, claimParams } from './shared/factories';
+import createVaultHelpers from './shared/vault';
+
+import { moveForwardTwoWeeks, generateNewAddress } from './shared';
+
+const { parseUnits } = ethers.utils;
+const { MaxUint256 } = ethers.constants;
+
+describe('Vault in async mode', () => {
+  let owner: SignerWithAddress;
+  let alice: SignerWithAddress;
+  let bob: SignerWithAddress;
+
+  let underlying: MockUST;
+  let vault: Vault;
+  let strategy: MockStrategyAsync;
+
+  let addUnderlyingBalance: (
+    account: SignerWithAddress,
+    amount: string,
+  ) => Promise<BigNumber>;
+  let addYieldToVault: (amount: string) => Promise<BigNumber>;
+
+  const TWO_WEEKS = BigNumber.from(time.duration.weeks(2).toNumber());
+
+  const TREASURY = generateNewAddress();
+  const PERFORMANCE_FEE_PCT = BigNumber.from('200');
+  const INVESTMENT_FEE_PCT = BigNumber.from('200');
+  const INVEST_PCT = BigNumber.from('9000');
+
+  const fixtures = deployments.createFixture(async ({ deployments }) => {
+    await deployments.fixture(['vaults']);
+
+    [owner] = await ethers.getSigners();
+
+    const ustDeployment = await deployments.get('UST');
+
+    underlying = MockUST__factory.connect(ustDeployment.address, owner);
+  });
+
+  beforeEach(() => fixtures());
+
+  beforeEach(async () => {
+    [owner, alice, bob] = await ethers.getSigners();
+
+    let Vault = await ethers.getContractFactory('Vault');
+    let MockStrategy = await ethers.getContractFactory('MockStrategyAsync');
+
+    vault = await Vault.deploy(
+      underlying.address,
+      TWO_WEEKS,
+      INVEST_PCT,
+      TREASURY,
+      owner.address,
+      PERFORMANCE_FEE_PCT,
+      INVESTMENT_FEE_PCT,
+      [],
+    );
+
+    underlying.connect(owner).approve(vault.address, MaxUint256);
+    underlying.connect(alice).approve(vault.address, MaxUint256);
+    underlying.connect(bob).approve(vault.address, MaxUint256);
+
+    strategy = await MockStrategy.deploy(vault.address, underlying.address);
+
+    await vault.connect(owner).setStrategy(strategy.address);
+
+    ({ addUnderlyingBalance, addYieldToVault } = createVaultHelpers({
+      vault,
+      underlying,
+    }));
+
+    await addUnderlyingBalance(alice, '1000');
+    await addUnderlyingBalance(bob, '1000');
+  });
+
+  describe("#claimYield doesn't rebalance the Vault's funds", () => {
+    it('reverts when yield > reserves', async () => {
+      const params = depositParams.build({
+        amount: parseUnits('100'),
+        inputToken: underlying.address,
+        claims: [claimParams.percent(100).to(alice.address).build()],
+      });
+      await vault.connect(alice).deposit(params);
+
+      await addYieldToVault('50');
+      await vault.connect(owner).updateInvested();
+
+      await expect(
+        vault.connect(alice).claimYield(alice.address),
+      ).to.be.revertedWith('VaultNotEnoughFunds');
+    });
+
+    it("doesn't revert and withdraws funds from the Vault when yield = reserves", async () => {
```

Comment:
`it("works if the yield = reserves")` ?

Reply:
[fixed](https://github.com/lindy-labs/sc_solidity-contracts/pull/136/commits/47be529313b023a0fe04a06247a0095ceba141c9)